draft-brunstrom-taps-impl.md

```diff
@@ -330,6 +343,32 @@ Protocol options are checked next in order. Whether or not a set of protocol, or
 
 Branching for derived endpoints is the final step, and may have multiple layers of derivation or resolution, such as DNS service resolution and DNS hostname resolution.
 
+
+## Ranking Branches
+
+Implementations SHOULD rank the branches of the tree of connection options in order to prioritize them for candidate racing. Leaf nodes on branches with higher rankings represent connection attempts that will be raced first. Implementations SHOULD order the branches to reflect the preferences expressed by the application for its new connection, including Protocol and Path Selection Properties as well as Application Intents, which are specified in {{draft-trammell-taps-interface}}. In addition to the properties provided by the application, an implementation MAY include additional criteria such as cached performance estimates, see {{performance-caches}}, or system policy, see {{role-of-system-policy}}, in the ranking.
+
+The Application Intents specified in {{draft-trammell-taps-interface}} may be used to rank branches in the following ways:
+
+* Capacity Profile:
+An implementation may use the Capacity Profile to prefer paths optimized for the application's expected traffic pattern:
+   * Interactive/Low Latency:
+     Prefer paths with the lowest expected latency
+   * Constant Rate:
+     Prefer paths that can satisfy the requested Stream Send or Stream Receive Bitrate
+   * Scavenger/Bulk:
+     Prefer paths with the highest expected available bandwidth
+
+* Size to be Sent or Received:
+An implementation may use the Size to be Sent or Received to compute an estimate of the completion time of a transfer over different available paths and then prefer the path with the shorter expected completion time. This Intent may be used instead of the Capacity profile, as the application does not always know whether its transfer will be latency-bound or bandwidth-bound, and thus may not be able to specify a Capacity Profile. A related paper is currently under submission.
```

Comment:
But again I'd like to know how you know about the path property - in this case, the expected completion time per path. Is this from previous measurements?

Reply:
Yes, we use cached performance estimates of Round Trip Time and expected maximum bandwidth to compute an estimated completion time on the paths. I modified the text to clarify this.